server/service/src/name/query.rs

```diff
@@ -20,9 +19,17 @@ pub fn get_names(
     let pagination = get_default_pagination(pagination, MAX_LIMIT, MIN_LIMIT)?;
     let repository = NameRepository::new(&ctx.connection);
 
-    let filter = filter
-        .unwrap_or_default()
-        .r#type(NameType::Patient.not_equal_to());
+    let filter = filter.unwrap_or_default();
+
+    let type_filter = match filter.r#type.clone() {
+        Some(filter_input) => EqualFilter::<NameType> {
+            not_equal_to: Some(NameType::Patient),
+            ..filter_input
```

Comment:
I guess this still isn't perfect, if you pass in a `ne` filter that would get overwritten. I tried to do a more comprehensive merge but it was looking unwieldy - any tips? https://github.com/msupply-foundation/open-msupply/compare/4079-facilities-list-should-only-include-stores...4079-facilities-list-should-only-include-stores(better-not-equal-filter)

Reply:
I'm not 100% sure of the intention here, It's basically a default filter to exclude names I think.

My suggestion is to make this more explicit in the API, but that will require some more testing for things that might want to include patients.
https://github.com/msupply-foundation/open-msupply/compare/4079-facilities-list-should-only-include-stores...4079-facilites-list-name-suggestion

I don't like this uncommented filter as the expected behaviour isn't entirely clear, is this a security protection thing? Is it a useful default? etc...